core/src/test/java/io/confluent/rest/ConnectionDurationFilterTest.java

```diff
@@ -0,0 +1,225 @@
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriBuilder;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+
+/*
+ *  Note on client support --
+ *  The Apache HttpClient library v4.x does not support HTTP/2.0, so we use
+ *  OkHttp for HTTP/2.0 tests.
+ */
+public class ConnectionDurationFilterTest {
+
+  static Server server;
+  private static final long MAX_CONNECTION_DURATION_MS = 100;
+
+  @BeforeAll
+  public static void setUp() throws Exception {
+    MyAddressApplication application =
+        new MyAddressApplication(
+            new MyAddressConfig(
+                ImmutableMap.of(
+                    "listeners", "http://localhost:8080",
+                    "http2.enabled", "true",
+                    "max.connection.duration.ms", String.valueOf(MAX_CONNECTION_DURATION_MS)))
+        );
+    server = application.createServer();
+    server.start();
+  }
+
+  @AfterAll
+  public static void tearDown() throws Exception {
+    server.stop();
+  }
+
+  @Test
+  public void connectionDurationTest_ApacheHttpClient_Ephemeral_HTTP1() throws Exception {
+
+    CloseableHttpClient client = createPersistentClient();
+    List<String> addressList = new ArrayList<>();
+
+    // Send a few requests with a delay between consecutive requests that is longer
+    // than the connection duration threshold.
+    // Each request should succeed and the server should return a different address for each request.
+    for (int i=0; i<5; i++) {
+
+      HttpGet request = createRequest(server.getURI());
+      assertEquals("HTTP/1.1", request.getProtocolVersion().toString());
+
+      CloseableHttpResponse response = client.execute(request);
+      assertEquals(Status.OK.getStatusCode(), response.getStatusLine().getStatusCode());
+
+      String responseBody = EntityUtils.toString(response.getEntity());
+      assertTrue(responseBody.startsWith("127.0.0.1:"));
+
+      response.close();
+      addressList.add(responseBody);
+      Thread.sleep(MAX_CONNECTION_DURATION_MS + 10);
+    }
+    client.close();
+
+    // check that we actually used 5 different addresses under the hood
+    assertEquals(5, addressList.stream().distinct().count());
+  }
+
+  @Test
+  public void connectionDurationTest_ApacheHttpClient_Persistent_HTTP1() throws Exception {
+
+    CloseableHttpClient client = createEphemeralClient();
```

Comment:
Naming should be consistent, similar to comment on line 83.

Reply:
Apologies. It's a typo.  will filx